plugins/circleci/src/circleci-config.ts

```diff
@@ -32,64 +50,103 @@ export default abstract class CircleCiConfigHook extends Hook {
-    const jobs = workflows?.['tool-kit']?.jobs
-    const nightlyJobs = workflows?.['nightly']?.jobs
+    if (!config) {
+      return false
+    }
+
+    const workflows = config.get('workflows') as YAML.YAMLMap
+    const jobs = workflows.getIn(['tool-kit', 'jobs']) as YAML.YAMLSeq
+    const nightlyJobs = workflows.getIn(['nightly', 'jobs']) as YAML.YAMLSeq
     if (!jobs || !nightlyJobs) {
       return false
     }
 
     // Automatically update tool kit orb if config is autogenerated
-    const rawConfig = await this.getCircleConfigRaw()
-    if (rawConfig && rawConfig.startsWith(automatedComment)) {
-      const currentOrb = config?.orbs['tool-kit']
+    if (isAutomatedConfig(config)) {
+      const currentOrb = config.getIn(['orbs', 'tool-kit']) as string
       const currentMajorTag = currentOrb && currentOrb.split('@')[1].split('.')[0]
 
       if (majorOrbVersion !== currentMajorTag) {
         return false
       }
     }
 
-    function hasJob(expectedJob: string, jobs: NonNullable<Workflow['jobs']>): boolean {
-      return jobs.some(
+    const hasJob = (expectedJob: string, jobs: YAML.YAMLSeq): boolean =>
+      jobs.items.some(
         (job) =>
-          (typeof job === 'string' && job === expectedJob) ||
-          (typeof job === 'object' && job.hasOwnProperty(expectedJob))
+          (YAML.isScalar(job) && job.value === expectedJob) || (YAML.isMap(job) && job.has(expectedJob))
       )
-    }
 
     return hasJob(this.job, jobs) && (!this.addToNightly || hasJob(this.job, nightlyJobs))
   }
 
-  async install(): Promise<void> {
-    const rawConfig = await this.getCircleConfigRaw()
-    if (rawConfig && !rawConfig.startsWith(automatedComment)) {
+  async install(state?: CircleCIState): Promise<CircleCIState> {
+    if (!state) {
+      state = {
+        jobs: {},
+        nightlyJobs: {},
+        runOnVersionTags: false
+      }
+    }
+
+    state.jobs[this.job] = this.jobOptions
+    if (this.addToNightly) {
+      state.nightlyJobs[this.job] = this.jobOptions
+    }
+    if (this.runOnVersionTags) {
+      state.runOnVersionTags = true
+    }
+    return state
+  }
+
+  async commitInstall(state: CircleCIState): Promise<void> {
+    const currentConfig = await this.getCircleConfig()
+    if (currentConfig && !isAutomatedConfig(currentConfig)) {
+      const formatMissingHooks = (hooks: string[]): string => hooks.map(styles.hook).join(', ')
       throw new ToolKitError(
-        `Please update your CircleCI config to include the \`${styles.hook(
-          this.job
-        )}\` job in the ${styles.heading('tool-kit')} workflow`
+        `Please update your CircleCI config to include the ${formatMissingHooks(
+          Object.keys(state.jobs)
+        )} jobs in the ${styles.heading('tool-kit')} workflow and the ${formatMissingHooks(
+          Object.keys(state.nightlyJobs)
+        )} jobs in the ${styles.heading('nightly')} workflow`
```

Comment:
I have been running this locally with the tool-kit sandbox. When I remove the automated config message from the config.yml file and also remove one job from the workflow, e.g. `tool-kit/test`:
```yml
- tool-kit/test:
          node-version: 16.14-browsers
          requires:
            - tool-kit/build
          filters:
            tags:
              only: /^v\d+\.\d+\.\d+(-.+)?/
```

and then run `npx dotcom-tool-kit --install` I get the full error message indicating that all jobs are missing - is that the intended behaviour?

<img width="713" alt="image" src="https://user-images.githubusercontent.com/17846996/201638130-77264311-2429-46a4-89e2-a7a4ecdb02a4.png">